### ISSUE TITLE
[bug] no-leaked-conditional-rendering and potential leaked empty string

### ISSUE THREAD

**Issue by mrudowski** [Rel1cx/eslint-react#853]
### Describe the bug

In docs we have example with empty string that breaks react native:

```
import React from "react";
 
function Example() {
  return <>{"" && <div />}</>;
  //        ^^
  //        - Possible unexpected value will be rendered (React Dom: renders nothing, React Native, with React below 18: crashes 💥).
}
```

So shouldn't `no-leaked-conditional-rendering` rule displays warning in this situations?
```
const someString = "";
{someString && <Something />}
```
```
const SampleComponent = ({someString}: {someString: string}) => {
  return (
    <div>
      {someString && <Something />}
    </div>
  )
}
```

### Expected behavior

I would expected warning
```
Potential leaked value someString that might cause unintentionally rendered values or rendering crashes.(@eslint-react/ no-leaked-conditional-rendering)
```

### Platform and versions

```Text
Node 20.14.0
@eslint-react/eslint-plugin: 1.16.1
```

**Comment by Rel1cx** [comment]
> Thanks for supporting `""` but... I think when we have a variable from props of type string, we cannot be sure if it is not `""`. So we should warning user and propose `!!anyStringVar` solution - the same as we do with `anyNumberVar`

Thanks for pointing this out, the new PR will fix this, along with proper version detection!
